service/storage_proxy.cc

```diff
@@ -122,6 +122,152 @@ seastar::metrics::label_instance current_scheduling_group_label() {
 
 }
 
+struct storage_proxy::remote {
```

Comment:
In the cover letter, you wrote a long explanation of what this "remote" is supposed to be, but in a few months from now, nobody will know where to find it. It's already hard enough to guess what a "storage_proxy" is and what really belongs inside it, and now you also have storage_proxy::remote as another mystery (isn't a "proxy" always about being "remote"?). Can you please add a comment before this class explaining its purpose, and what belongs in it (and what doesn't)?

Reply:
> (isn't a "proxy" always about being "remote"?)

"Proxy" is about been in between. The `storage_proxy` is between a transport and the storage.

Reply:
> Can you please add a comment before this class explaining its purpose, and what belongs in it (and what doesn't)?

I will